src/intentManager.js

```diff
@@ -51,167 +55,98 @@ define([
+                tag = _.findWhere(unmappedIntentTags, {nodeID: nodeID});
 
-        widget.getControl = function () {
-            if (widget.isDisabled()) {
-                input.prop('disabled', true);
+            if (!tag) {
+                tag = makeODKXIntentTag(nodeID, null);
             }
-            return input;
-        };
-
-        widget.setValue = function (value) {
-            input.val(value);
-        };
 
-        widget.getValue = function() {
-            return input.val();
-        };
-
-        widget.updateValue = function () {
-            widget.mug.intentTag.setAttr('path', widget.getValue());
-        };
+            _.each(INTENT_SPECIFIC_SPECS, function (key) {
+                mug.p[key] = tag[key];
+            });
 
-        input.bind("change keyup", widget.updateValue);
-        return widget;
+            delete unmappedIntentTags[tag.nodeID];
+        }
     }
 
-    function androidIntentExtra(mug, options) {
-        options.id = "intent-extra";
-        var widget = widgets.baseKeyValue(mug, options);
-        widget.currentValue = (mug.intentTag) ? mug.intentTag.getAttr('extra') : {};
-        widget.definition = mug.p.getDefinition('androidIntentExtra');
+    function writeIntentXML (xmlWriter, tree) {
+        // make sure any leftover intent tags are still kept
+        _.each(unmappedIntentTags, function (tag) {
+            writeXML(xmlWriter, tag);
+        });
 
-        widget.save = function () {
-            if (widget.mug.intentTag) {
-                widget.mug.intentTag.setAttr('extra', widget.getValidValues());
+        var intents,
+        getIntentMugs = function(node) {
+            var mug = node.getValue();
+            if (!mug || node.isRootNode) {
+                return null;
+            }
+            if (mug.options.dataType === 'intent') {
+                return mug;
+            } else {
+                return null;
             }
         };
+        intents = tree.treeMap(getIntentMugs);
+        intents.map(function (intentMug) {
+            writeXML(xmlWriter, intentMug.p);
+        });
+    }
 
+    function androidIntentAppId(mug, options) {
+        var widget = widgets.text(mug, options),
+            input = widget.input;
+        input.attr('placeholder', 'Insert Android Application ID');
         return widget;
     }
 
-    function androidIntentResponse(mug, options) {
-        options.id = "intent-response";
-        var widget = widgets.baseKeyValue(mug, options);
-        widget.currentValue = (mug.intentTag) ? mug.intentTag.getAttr('response') : {};
-        widget.definition = mug.p.getDefinition('androidIntentResponse');
-
-        widget.save = function () {
-            if (widget.mug.intentTag) {
-                widget.mug.intentTag.setAttr('response', widget.getValidValues());
-            }
-        };
-
-        return widget;
+    function serializeAttrs(value, key, mug, data) {
+        data[key] = _.omit(mug.p[key], "");
```

Comment:
subtle change here: previously the `""` key was dropped if it's value was also `""`. Now it is unconditionally dropped. Is that intentional?